### PR TITLE
Document and test XJC plugin options

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

**Comment by ppalaga** [comment]
Would you like to review, @llowinge?

File: extensions/xjc-plugins/runtime/src/main/doc/intro.adoc [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-* `-xjc-Xjavadoc`
-* `-xjc-Xpl`
-* `-xjc-Xts`
-* `-xjc-Xwsdlextension`
+* `https://cxf.apache.org/cxf-xjc-boolean.html[-xjc-Xbg]` - generate `getFoo()` instead of `isFoo()` accessor methods for boolean fields.
+* `https://cxf.apache.org/cxf-xjc-dv.html[-xjc-Xdv]` - let the generated getter methods return the default value defined in the schema unless the field is set explicitly.
+* `-xjc-Xjavadoc` - generate JavaDoc based on `xs:documentation` present in the schema.
+* `-xjc-Xproperty-listener` - add `PropertyChangeListener` support to the generated beans.
+* `https://cxf.apache.org/cxf-xjc-ts.html[-xjc-Xts]` - generate `toString()` methods in model classes.
+* `https://cxf.apache.org/cxf-xjc-wsdlextension.html[-xjc-Xwsdlextension]` - generate beans that can be used directly with WSDL4J as extensors in the WSDL.
```
Comment (ppalaga) [inline extensions/xjc-plugins/runtime/src/main/doc/intro.adoc:10]: @llowinge this is the main docs change. As you can see I have not found suitable CXF pages for all options. 
Some of the options were bogus, perhaps some old names that do not work now. I better added a test below to make sure that the spelling of the option names is correct. In other words, thanks for complaining @llowinge!

File: integration-tests/wsdl2java/src/main/resources/application.properties [hunk 2]
```diff
@@ -1,2 +1,2 @@
 quarkus.cxf.codegen.wsdl2java.includes = **.wsdl
-quarkus.cxf.codegen.wsdl2java.additional-params = -wsdlLocation,classpath:wsdl/CalculatorService.wsdl,-xjc-Xts
+quarkus.cxf.codegen.wsdl2java.additional-params = -wsdlLocation,classpath:wsdl/CalculatorService.wsdl,-xjc-Xbg,-xjc-Xdv,-xjc-Xjavadoc,-xjc-Xproperty-listener,-xjc-Xts,-xjc-Xwsdlextension
```
Comment (ppalaga) [inline integration-tests/wsdl2java/src/main/resources/application.properties:2]: `-xjc-Xdv` is not covered by any assertion in the test below. I wanted to use the default value of 42 in Result.result, but then, after I published the test container image 1.3, I found that it does not work for primitive types. 
But at least, it is passed to wsdl2java and wsdl2java does not complain.

**Comment by ppalaga** [comment]
62d0ba3: `-xjc-Xdv` now has a test

File: docs/modules/ROOT/pages/user-guide/generate-java-from-wsdl.adoc [hunk 3]
```diff
@@ -59,5 +59,5 @@ quarkus.cxf.codegen.wsdl2java.bar-params.additional-params = -wsdlLocation,wsdl/
 ----
 
 TIP: Add `io.quarkiverse.cxf:quarkus-cxf-xjc-plugins` dependency to your project to be able to use
```
Comment (llowinge) [inline docs/modules/ROOT/pages/user-guide/generate-java-from-wsdl.adoc:61]: Can you add link to https://quarkiverse.github.io/quarkiverse-docs/quarkus-cxf/1.5/reference/extensions/quarkus-cxf-xjc-plugins.html somewhere in the end of this TIP ?

**Pull request merged**